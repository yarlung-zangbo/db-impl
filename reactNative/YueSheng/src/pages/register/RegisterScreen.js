/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {StyleSheet, Text, KeyboardAvoidingView, TouchableOpacity, View, ImageBackground, TextInput} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import {themeColor, image, width, height} from '../variable/Commen'
export default class LoginScreen extends Component<Props> {
    render() {
        return (
            <ImageBackground source={require("YueSheng/src/image/b.jpg")} style={{width: '100%', height: '100%'}}>
                <View  style={styles.container}>
                    <KeyboardAvoidingView style={styles.avoid}>
                        <View>
                            <View style={{alignItems:'center'}}>
                                <FontAwesome name={"user-circle"}
                                             style={{color: "#000", fontSize:50}}/>
                            </View>
                            <View style={styles.loginView}>
                                <View >
                                    <TextInput style={styles.input}
                                               placeholder={"Username"}
                                               placeholderTextColor={"#888"}/>
                                </View>
                                <View>
                                    <TextInput style={styles.input}
                                               placeholder={"Email"}
                                               placeholderTextColor={"#888"}/>
                                </View>
                                <View>
                                    <TextInput style={styles.input}
                                               placeholder={"Password"}
                                               placeholderTextColor={"#888"}/>
                                </View>
                                <View>
                                    <TextInput style={styles.input}
                                               placeholder={"Confirm Password"}
                                               placeholderTextColor={"#888"}/>
                                </View>
                                <TouchableOpacity>
                                    <View style={styles.registerIcon}>
                                        <Text style={{color:"#fff", fontSize:15}}> 注册 </Text>
                                    </View>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </KeyboardAvoidingView>
                </View>
            </ImageBackground>
        );
    }
}
const b=image+"b.jpg";

const styles = StyleSheet.create({
    container: {
        flex:1,
        alignItems: 'center',
        backgroundColor: "rgba(255, 255, 255, 0.4)",
        justifyContent: 'center',
    },
    avoid:{
        alignItems:'center',
        flex:4,
        justifyContent:"space-around",
    },
    loginView:{
        paddingTop:20,
    },
    input:{
        width:width*0.7,
        height:40,
        backgroundColor: 'rgba(0, 0, 0, 0.6)',
        paddingLeft:10,
        paddingRight:10,
        marginBottom: 20,
        borderRadius:6,
        color:'#fff',
    },
    registerIcon:{
        backgroundColor: "rgba(243,75,89, 0.9)",
        opacity:0.95,
        borderRadius: 6,
        height:40,
        justifyContent:'center',
        alignItems:'center',
    }
});
