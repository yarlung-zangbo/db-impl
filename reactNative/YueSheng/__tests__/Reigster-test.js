/**
 * @format
 */

import 'react-native';
import React from 'react';
import RegisterScreen from '../src/pages/register/RegisterScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<RegisterScreen />);
    expect(tree).toMatchSnapshot();
});
