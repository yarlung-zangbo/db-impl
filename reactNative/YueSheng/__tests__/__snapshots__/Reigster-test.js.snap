// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/image/b.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.4)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 4,
          "justifyContent": "space-around",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#000",
                  "fontSize": 50,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(234,75, 89)",
              }
            }
          >
             
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingTop": 20,
            }
          }
        >
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Username"
              placeholderTextColor="#888"
              rejectResponderTermination={true}
              selectionColor="rgb(234,75, 89)"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.6)",
                  "borderRadius": 6,
                  "color": "#fff",
                  "height": 40,
                  "marginBottom": 20,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "width": 525,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Email"
              placeholderTextColor="#888"
              rejectResponderTermination={true}
              selectionColor="rgb(234,75, 89)"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.6)",
                  "borderRadius": 6,
                  "color": "#fff",
                  "height": 40,
                  "marginBottom": 20,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "width": 525,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#888"
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="rgb(234,75, 89)"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.6)",
                  "borderRadius": 6,
                  "color": "#fff",
                  "height": 40,
                  "marginBottom": 20,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "width": 525,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Confirm Password"
              placeholderTextColor="#888"
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="rgb(234,75, 89)"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.6)",
                  "borderRadius": 6,
                  "color": "#fff",
                  "height": 40,
                  "marginBottom": 20,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "width": 525,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(243,75,89, 0.9)",
                  "borderRadius": 6,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 0.95,
                  "width": 525,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                  }
                }
              >
                 注册 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
